amxmodx/file.cpp

```diff
@@ -7,484 +7,503 @@
 
-	return 0;
+	return read / size;
 }
 
+// native fwrite(file, data, mode);
 static cell AMX_NATIVE_CALL amx_fwrite(AMX *amx, cell *params)
 {
-	FILE *fp = (FILE *)params[1];
+	FileObject* fp = reinterpret_cast<FileObject*>(params[1]);
 
 	if (!fp)
+	{
 		return 0;
+	}
+
+	cell   data = params[2];
+	size_t size = params[3];
 
-	size_t mode = params[3];
-	switch (mode)
+	switch (size)
 	{
-	case 1:
+		case BLOCK_CHAR:
 		{
-			char a = static_cast<char>(params[2]);
-			return fwrite(&a, sizeof(char), 1, fp);
+			char value = static_cast<char>(data);
+			return fp->Write(&value, sizeof(value));
 		}
-	case 2:
+		case BLOCK_SHORT:
 		{
-			short b = static_cast<short>(params[2]);
-			return fwrite(&b, sizeof(short), 1, fp);
+			short value = static_cast<short>(data);
+			return fp->Write(&value, sizeof(value));
 		}
-	case 4:
+		default:
+		case BLOCK_INT:
 		{
-			int c = static_cast<int>(params[2]);
-			return fwrite(&c, sizeof(int), 1, fp);
+			int value = static_cast<int>(data);
+			return fp->Write(&value, sizeof(value));
 		}
 	}
 
 	return 0;
 }
 
+// native fwrite_raw(file, const stream[], blocksize, mode);
 static cell AMX_NATIVE_CALL amx_fwrite_raw(AMX *amx, cell *params)
 {
-	FILE *fp = (FILE *)params[1];
+	FileObject* fp = reinterpret_cast<FileObject*>(params[1]);
 
 	if (!fp)
+	{
 		return 0;
+	}
 
-	cell *addr = get_amxaddr(amx, params[2]);
-	return fwrite(addr, params[3], params[4], fp);
+	cell* data = get_amxaddr(amx, params[2]);
+
+	return fp->Write(&data, params[3] * params[4]);
 }
 
+// native fread_raw(file, stream[], blocksize, blocks);
 static cell AMX_NATIVE_CALL amx_fread_raw(AMX *amx, cell *params)
 {
-	FILE *fp = (FILE *)params[1];
+	FileObject* fp = reinterpret_cast<FileObject*>(params[1]);
 
 	if (!fp)
+	{
 		return 0;
+	}
+
+	cell* data = get_amxaddr(amx, params[2]);
 
-	cell *addr = get_amxaddr(amx, params[2]);
-	size_t size = static_cast<cell>(params[3]);
-	size_t blocks = static_cast<cell>(params[4]);
-    
-	return fread(addr, size, blocks, fp);
+	return fp->Read(data, params[3] * params[4]);
 }
 
+// native fread(file, &data, mode);
 static cell AMX_NATIVE_CALL amx_fread(AMX *amx, cell *params)
 {
-	FILE *fp = (FILE *)params[1];
+	FileObject* fp = reinterpret_cast<FileObject*>(params[1]);
 
 	if (!fp)
+	{
 		return 0;
+	}
+
+	cell *data = get_amxaddr(amx, params[2]);
 
-	cell *addr = get_amxaddr(amx, params[2]);
 	switch (params[3])
 	{
-	case 1:	//char
+		case BLOCK_CHAR:
 		{
-			char a;
-			size_t res = fread(&a, sizeof(char), 1, fp);
-			*addr = static_cast<cell>(a);
+			char value;
+			size_t res = fp->Read(&value, sizeof(value));
+			*data = static_cast<cell>(value);
 			return res;
 		}
-	case 2:	//short
+		case BLOCK_SHORT:
 		{
-			short a;
-			size_t res = fread(&a, sizeof(short), 1, fp);
-			*addr = static_cast<cell>(a);
+			short value;
+			size_t res = fp->Read(&value, sizeof(value));
+			*data = static_cast<cell>(value);
 			return res;
 		}
-	case 4:	//int
-	default:
+		case BLOCK_INT:
+		default:
```

Comment:
... I say as the old code does the same thing. Can we retroactively remove this...?